p2p/communication.go

```diff
@@ -173,13 +173,14 @@ func (c *Communication) readFromStream(stream network.Stream) {
 			return
 		}
 		c.logger.Debug().Msgf(">>>>>>>[%s] %s", wrappedMsg.MessageType, string(wrappedMsg.Payload))
-		c.streamMgr.AddStream(wrappedMsg.MsgID, stream)
 		channel := c.getSubscriber(wrappedMsg.MessageType, wrappedMsg.MsgID)
 		if nil == channel {
 			c.logger.Debug().Msgf("no MsgID %s found for this message", wrappedMsg.MsgID)
 			c.logger.Debug().Msgf("no MsgID %s found for this message", wrappedMsg.MessageType)
+			_ = stream.Close()
```

Comment:
Is the stream closed elsewhere in the code after returning from this function? Closing the stream in a function that is called `readFromStream` seems like a logic breaking change. IMHO the stream should be closed somewhere upstream.

Reply:
I'm just backporting the original change: https://github.com/zeta-chain/go-tss/commit/fb3ceba7415115fe46d1f66927166ad0193192e9.

It appears that the stream would just be leaked and held open since the `StreamMgr` typically holds a reference and manages the stream. This change just closes the stream if it is unexpected.

Overall there is a lot of suspect logic in `go-tss` which would be great to examine and/or fully rewrite if we get the chance.